include/range/v3/range_access.hpp

```diff
@@ -162,19 +162,19 @@ namespace ranges
                 end.equal(pos)
             )
             template<typename Cur>
-            static auto prev(Cur & pos)
+            static RANGES_CXX14_CONSTEXPR auto prev(Cur & pos)
             RANGES_DECLTYPE_AUTO_RETURN
             (
                 pos.prev()
             )
             template<typename Cur, typename D>
-            static auto advance(Cur & pos, D n)
+            static RANGES_CXX14_CONSTEXPR auto advance(Cur & pos, D n)
             RANGES_DECLTYPE_AUTO_RETURN
             (
                 pos.advance(n)
             )
-            template<typename Cur>
-            static auto distance_to(Cur const &pos0, Cur const &pos1)
+            template<typename Cur0, typename Cur1>
+            static RANGES_CXX14_CONSTEXPR auto distance_to(Cur0 const &pos0, Cur1 const &pos1)
```

Comment:
I fixed this problem in `basic_iterator`. Apologies for the merge conflicts. :-(